plugin-testing-framework/script/test.ts

```diff
@@ -0,0 +1,470 @@
+    
+    /**
+     * testName - used as the expectation in the call to it
+     * test - the test to provide to it
+     * isCoreTest - whether or not the test should run when "--core" is supplied
+     * only - if true, use it.only
+     */
+    constructor(testName: string, test: (projectManager: tm.ProjectManager, targetPlatform: platform.IPlatform, done: MochaDone) => void, isCoreTest: boolean, options?: { only?: boolean, skip?: boolean }) {
+        super(options);
+        
+        this.testName = testName;
+        this.test = test;
+        this.isCoreTest = isCoreTest;
+    }
+    
+    /**
+     * Called to create the test suite by the initializeTests function
+     * 
+     * coreTestsOnly - Whether or not only core tests are to be run
+     * projectManager - The projectManager instance that these tests are being run with
+     * targetPlatform - The platform that these tests are going to be run on
+     */
+    public create(coreTestsOnly: boolean, projectManager: tm.ProjectManager, targetPlatform: platform.IPlatform): void {
+        if (!this.skip && (!coreTestsOnly || this.isCoreTest)) {
+            it(this.testName, this.test.bind(this, projectManager, targetPlatform));
+        }
+    }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+// Use these functions in tests
+
+/**
+ * Returns a default empty response to give to the app in a checkForUpdate request
+ */
+export function createDefaultResponse(): su.CheckForUpdateResponseMock {
+    var defaultResponse = new su.CheckForUpdateResponseMock();
+
+    defaultResponse.downloadURL = "";
+    defaultResponse.description = "";
+    defaultResponse.isAvailable = false;
+    defaultResponse.isMandatory = false;
+    defaultResponse.appVersion = "";
+    defaultResponse.packageHash = "";
+    defaultResponse.label = "";
+    defaultResponse.packageSize = 0;
+    defaultResponse.updateAppVersion = false;
+
+    return defaultResponse;
+}
+
+/**
+ * Returns a default update response to give to the app in a checkForUpdate request
+ */
+export function createUpdateResponse(mandatory: boolean = false, targetPlatform?: platform.IPlatform, randomHash: boolean = true): su.CheckForUpdateResponseMock {
+    var updateResponse = new su.CheckForUpdateResponseMock();
+    updateResponse.isAvailable = true;
+    updateResponse.appVersion = "1.0.0";
+    updateResponse.downloadURL = "mock.url/download";
+    updateResponse.isMandatory = mandatory;
+    updateResponse.label = "mock-update";
+    updateResponse.packageHash = "12345-67890";
+    updateResponse.packageSize = 12345;
+    updateResponse.updateAppVersion = false;
+
+    if (!!targetPlatform) updateResponse.downloadURL = targetPlatform.getServerUrl() + "/download";
+    
+    // We need unique hashes to avoid conflicts.
+    if (randomHash) {
+        updateResponse.packageHash = "randomHash-" + Math.floor(Math.random() * 10000);
+    }
+    
+    return updateResponse;
+}
```

Comment:
I feel like maybe the `createXXResponse` methods can go to the server util file.